airflow/jobs/scheduler_job.py

```diff
@@ -534,42 +535,63 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
         session.commit()
 
     @provide_session
-    def execute_on_failure_callbacks(
+    def execute_callbacks(
         self,
         dagbag: DagBag,
-        failure_callback_requests: List[FailureCallbackRequest],
+        callback_requests: List[CallbackRequest],
         session: Session = None
     ) -> None:
         """
         Execute on failure callbacks. These objects can come from SchedulerJob or from
         DagFileProcessorManager.
 
-        :param failure_callback_requests: failure callbacks to execute
-        :type failure_callback_requests: List[airflow.utils.dag_processing.FailureCallbackRequest]
+        :param dagbag: Dag Bag of dags
+        :param callback_requests: failure callbacks to execute
+        :type callback_requests: List[airflow.utils.callback_requests.CallbackRequest]
         :param session: DB session.
         """
-        for request in failure_callback_requests:
-            simple_ti = request.simple_task_instance
-            if simple_ti.dag_id in dagbag.dags:
-                dag = dagbag.dags[simple_ti.dag_id]
-                if simple_ti.task_id in dag.task_ids:
-                    task = dag.get_task(simple_ti.task_id)
-                    ti = TI(task, simple_ti.execution_date)
-                    # Get properties needed for failure handling from SimpleTaskInstance.
-                    ti.start_date = simple_ti.start_date
-                    ti.end_date = simple_ti.end_date
-                    ti.try_number = simple_ti.try_number
-                    ti.state = simple_ti.state
-                    ti.test_mode = self.UNIT_TEST_MODE
-                    ti.handle_failure(request.msg, ti.test_mode, ti.get_template_context())
-                    self.log.info('Executed failure callback for %s in state %s', ti, ti.state)
+        for request in callback_requests:
+            if isinstance(request, TaskCallbackRequest) and request.is_failure_callback:
```

Comment:
Why the `is_failure_callback` check here?

Reply:
Currently, we only run failure callbacks (including Zombies) via this function. Handle callbacks are directly taken care in 

- https://github.com/astronomer/airflow/blob/f0d64d44e82eef4cb4a5dbd85969a44de3939b99/airflow/jobs/local_task_job.py#L170
- https://github.com/astronomer/airflow/blob/f0d64d44e82eef4cb4a5dbd85969a44de3939b99/airflow/models/taskinstance.py#L1131


Moved this check inside _execute_task_callbacks